ui.apps/src/main/content/jcr_root/apps/acs-commons/config/org.apache.sling.serviceusermapping.impl.ServiceUserMapperImpl.amended-acs-commons.xml

```diff
@@ -0,0 +1,33 @@
+          user.mapping="[
+                        com.adobe.acs.acs-aem-commons-bundle:ensure-oak-index=[acs-commons-ensure-oak-index-service],
+                        com.adobe.acs.acs-aem-commons-bundle:email-service=[acs-commons-email-service],
+                        com.adobe.acs.acs-aem-commons-bundle:httpcache-jcr-storage-service=[acs-commons-httpcache-jcr-storage-service],
+                        com.adobe.acs.acs-aem-commons-bundle:review-task-asset-mover=[acs-commons-review-task-asset-mover-service],
+                        com.adobe.acs.acs-aem-commons-bundle:error-page-handler=[acs-commons-error-page-handler-service],
+                        com.adobe.acs.acs-aem-commons-bundle:form-helper=[acs-commons-form-helper-service],
+                        com.adobe.acs.acs-aem-commons-bundle:dispatcher-flush=[acs-commons-dispatcher-flush-service],
+                        com.adobe.acs.acs-aem-commons-bundle:package-replication-status-event-listener=[acs-commons-package-replication-status-event-service],
+                        com.adobe.acs.acs-aem-commons-bundle:component-error-handler=[acs-commons-component-error-handler-service],
+                        com.adobe.acs.acs-aem-commons-bundle:system-notifications=[acs-commons-system-notifications-service],
+                        com.adobe.acs.acs-aem-commons-bundle-twitter:twitter-updater=[acs-commons-twitter-updater-service],
+                        com.adobe.acs.acs-aem-commons-bundle:workflow-remover=[acs-commons-workflow-remover-service],
+                        com.adobe.acs.acs-aem-commons-bundle:bulk-workflow=[acs-commons-bulk-workflow-service],
+                        com.adobe.acs.acs-aem-commons-bundle:bulk-workflow-runner=[workflow-process-service],
+                        com.adobe.acs.acs-aem-commons-bundle:ensure-service-user=[acs-commons-ensure-service-user-service],
+                        com.adobe.acs.acs-aem-commons-bundle:shared-component-props=[acs-commons-shared-component-props-service],
+                        com.adobe.acs.acs-aem-commons-bundle:manage-controlled-processes=[acs-commons-manage-controlled-processes-service],
+                        com.adobe.acs.acs-aem-commons-bundle:automatic-package-replicator=[acs-commons-automatic-package-replicator-service],
+                        com.adobe.acs.acs-aem-commons-bundle:on-deploy-scripts=[acs-commons-on-deploy-scripts-service],
+                        com.adobe.acs.acs-aem-commons-bundle:remote-assets=[acs-commons-remote-assets-service],
+                        com.adobe.acs.acs-aem-commons-bundle:workflowpackagemanager-service=[acs-commons-workflowpackagemanager-service],
+                        com.adobe.acs.acs-aem-commons-bundle:redirect-manager=[acs-commons-manage-redirects-service],
+                        com.adobe.acs.acs-aem-commons-bundle:file-fetch=[acs-commons-file-fetch-service]
+                    ]"/>
+
+<!-- sling:OsgiConfig format used due to 6.4.8 support -->
```

Comment:
Even AEM 6.4 supports the better suited .config format (https://sling.apache.org/documentation/bundles/configuration-installer-factory.html#configuration-files-config). I would not recommend adding new configuration with sling:OsgiConfig format which has a lot of flaws: https://sling.apache.org/documentation/bundles/configuration-installer-factory.html#slingosgiconfig-resources

Reply:
Are these any caveats with .config? That's what we had before and when installed no service user mappings would get registered. Moving them back to sling:OsgiConfig and they work just fine.  The .config did work on 6.5 and AEM CS though... thoughts on why 6.4 was problematic?

Reply:
Not that I know of.  In case there is issues with .config on 6.4.8 this is probably due to escaping. Any logs available from OSGi installer with the problematic .config?

Reply:
Ok, the problem is not the .config, but the filename. Tilde as separator is only supported since Configuration Factory Packages 1.2.0 (https://issues.apache.org/jira/browse/SLING-7786) while AEM 6.4 still ships with 1.1.x. So just converting the filename from `ui.apps/src/main/content/jcr_root/apps/acs-commons/config/org.apache.sling.serviceusermapping.impl.ServiceUserMapperImpl.amended~acs-commons.config` to `ui.apps/src/main/content/jcr_root/apps/acs-commons/config/org.apache.sling.serviceusermapping.impl.ServiceUserMapperImpl.amended-acs-commons.config` should be enough to fix this.

Reply:
sunuva. didnt even see that .. 👓  ...
Ill flip it back and update to `-`..